packages/react-reconciler/src/ReactFiberThrow.new.js

```diff
@@ -444,6 +444,10 @@ function throwException(
     }
   }
 
+  if (getIsHydrating() && sourceFiber.mode & ConcurrentMode) {
+    markDidSuspendWhileHydratingDEV();
```

Comment:
Nit: Move this to the `else` block on line 546 so we don't check these twice in the Suspense case